NMS-9415: Check if ipAddressIfIndex contains any data before attempting to parse it

brynjare opened the pull request:
OpenNMS throws a NullPointerException during the node scan of devices with non-existent IP-MIB::ipAddressIfIndex entries.

This is a quick bugfix / workaround, which simply checks if anything was returned in the SNMP reply. If the reply was empty, a warning will be printed before returning null.

* JIRA: https://issues.opennms.org/browse/NMS-9415

opennms-bamboo commented:
Welcome @brynjare.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @brynjare.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

brynjare commented:
Just wanted to add that I have every intention to sign the OCA, but I'll need to get it approved before I can proceed. I'll submit it as soon as possible.

brynjare commented:
Thanks for the feedback, Jeff! The OCA is finally signed and submitted by email.

indigo423 commented:
Ulf can you please recheck the contributors OCA?

indigo423 commented:
#1549 went green in bamboo. So 👍 for me. Let me

brynjare commented:
Just changing the merge base didn't work, as it would include the bump from v20 to v21, in eea3314.
I'll try to figure out a way to fix this, but feel free to add any suggestions on how to best fix this.

Sorry for creating a mess of an otherwise simple pull request.

brynjare commented:
Changed the original commit to be based on 329e481 instead, and updated the base branch to release-20.0.1. Hopefully this should solve it, but feel free to let me know if anything looks wonky with the pull request.
Thank you for considering the bugfix for the next release, and for all your helpful feedback!

mvrueden commented:
Manually cherry-picked into https://github.com/OpenNMS/opennms/tree/jira/NMS-9415 and re-targeted it to `foundation`(-2015). Let's wait for bamboo and then manually merge it.

mvrueden commented:
Closing in favor of https://github.com/OpenNMS/opennms/pull/1552. Thank you
